cli/src/pcluster/config/cluster_config.py

```diff
@@ -597,7 +597,7 @@ def __init__(
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.secured = Resource.init_param(secured)
+        self.secured = Resource.init_param(secured, default=True)
```

Comment:
i'm pretty sure this is the change that nullifies my paranoid comment regarding the setting of the IMDS secured cookbook attribute?

Reply:
Yes, this ensures that Imds.Secured cannot be None. 
I think the only place where it make sense to enforce that it must not be None is inside the validator.
So, I think we can assume it is not None when setting the dns.json